### PR TITLE
TINKERPOP-2723 Make graphbinary default in gremlin-python and dotnet

### PR DESCRIPTION
**Pull request by vkagamlyk**
https://issues.apache.org/jira/browse/TINKERPOP-2723

Make graphbinary default in gremlin-python and dotnet

### CODE REVIEW

**Comment by lyndonbauto** [comment]
I am going to hold off reviewing this until #1676 is merged.

**Comment by lyndonbauto** [comment]
VOTE+1

**Review by FlorianHockmann — CHANGES_REQUESTED** [review]
The code changes look good to me, but could you please also update the docs and add a note to the upgrade docs and a changelog entry as this is a breaking change?

[This section about configuration for gremlin-python](https://tinkerpop.apache.org/docs/current/reference/#gremlin-python-configuration) currently mentions `GraphSONMessageSerializer` as the default serializer
and [this section shows for Gremlin.Net how to change the default serializer](https://tinkerpop.apache.org/docs/current/reference/#gremlin-dotnet-serialization) from GraphSON 3 to GraphBinary / GraphSON 2.

(I've added a link to the JIRA issue to the PR description.)

**Comment by vkagamlyk** [comment]
> The code changes look good to me, but could you please also update the docs and add a note to the upgrade docs and a changelog entry as this is a breaking change?
> 
> [This section about configuration for gremlin-python](https://tinkerpop.apache.org/docs/current/reference/#gremlin-python-configuration) currently mentions `GraphSONMessageSerializer` as the default serializer and [this section shows for Gremlin.Net how to change the default serializer](https://tinkerpop.apache.org/docs/current/reference/#gremlin-dotnet-serialization) from GraphSON 3 to GraphBinary / GraphSON 2.
> 
> (I've added a link to the JIRA issue to the PR description.)

Doc's updated

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Docs/Reference/GremlinVariantsTests.cs
Comment (FlorianHockmann) [inline gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Docs/Reference/GremlinVariantsTests.cs:82]: Not that important, but you could also remove the test `SerializationGraphBinaryTest` above now as it isn't used any more in docs.

**Review by FlorianHockmann — APPROVED** [review]
I added one more nitpick comment, but this is good to go from my side with or without addressing it.

[A changelog entry](https://github.com/apache/tinkerpop/blob/master/CHANGELOG.asciidoc) and a note in the [upgrade docs](https://github.com/apache/tinkerpop/blob/master/docs/src/upgrade/release-3.6.x.asciidoc) is also still missing, but whoever merges this can also do that while merging, I guess.

VOTE +1

**Pull request merged**
